sky/utils/common_utils.py

```diff
@@ -485,14 +486,17 @@ def read_yaml_all(path: str) -> List[Dict[str, Any]]:
         return read_yaml_all_str(f.read())
 
 
-def dump_yaml(path: str, config: Union[List[Dict[str, Any]],
-                                       Dict[str, Any]]) -> None:
+def dump_yaml(path: str,
+              config: Union[List[Dict[str, Any]], Dict[str, Any]],
+              create_if_not_exists: bool = False) -> None:
     """Dumps a YAML file.
 
     Args:
         path: the path to the YAML file.
         config: the configuration to dump.
     """
+    if create_if_not_exists:
+        pathlib.Path(path).touch()
```

Comment:
```suggestion
        pathlib.Path(path).touch(exist_ok=True)
```
This is the default, but let's make it explicit for readability.